packages/main/src/Table.js

```diff
@@ -540,6 +548,106 @@ class Table extends UI5Element {
 		if (isAltUp || isDownAlt(event)) {
 			return this._handleArrowAlt(isAltUp, event.target);
 		}
+
+		if ((isUpShift(event) || isDownShift(event)) && this.isMultiSelect) {
+			this._handleArrowNav(event);
+		}
+
+		if (isHomeCtrl(event)) {
+			event.preventDefault();
+
+			this._itemNavigation._handleHome(event);
+			this._itemNavigation._applyTabIndex();
+			this._itemNavigation._focusCurrentItem();
+		}
+
+		if (isEndCtrl(event)) {
+			event.preventDefault();
+
+			this._itemNavigation._handleEnd(event);
+			this._itemNavigation._applyTabIndex();
+			this._itemNavigation._focusCurrentItem();
+		}
+
+		if ((isHomeShift(event) || isEndShift(event)) && this.isMultiSelect) {
+			this._handleHomeEndSelection(event);
+		}
+	}
+
+	_handleArrowNav(event) {
+		const isRowFocused = this._getActiveElementTagName() === "ui5-table-row";
+
+		if (!isRowFocused) {
+			return;
+		}
+
+		const previouslySelectedRows = this.selectedRows;
+		const currentItem = this.currentItem;
+		const currentItemIdx = this.currentItemIdx;
+
+		const prevItemIdx = currentItemIdx - 1;
+		const nextItemIdx = currentItemIdx + 1;
+
+		const prevItem = this.rows[prevItemIdx];
+		const nextItem = this.rows[nextItemIdx];
+		const wasSelected = currentItem.selected;
+
+		if ((isUpShift(event) && !prevItem) || (isDownShift(event) && !nextItem)) {
+			return;
+		}
+
+		if (isUpShift(event)) {
+			currentItem.selected = currentItem.selected && !prevItem.selected;
+			prevItem.selected = currentItem.selected || (wasSelected && !currentItem.selected);
+
+			prevItem.focus();
+		}
+
+		if (isDownShift(event)) {
+			currentItem.selected = currentItem.selected && !nextItem.selected;
+			nextItem.selected = currentItem.selected || (wasSelected && !currentItem.selected);
+
+			nextItem.focus();
+		}
+
+		const selectedRows = this.selectedRows;
+
+		this.fireEvent("selection-change", {
+			selectedRows,
+			previouslySelectedRows,
+		});
+	}
+
+	_handleHomeEndSelection(event) {
+		const isRowFocused = this._getActiveElementTagName() === "ui5-table-row";
```

Comment:
This condition won't work always. If have the "scoping" feature turned on, the ui5-table-row tag might be renamed to myprefixed-ui5-table-row by the application developers for the use-case of multiple version in micro-frontend architecture.

 Furthermore, for the OpenUI5 enablement, we machine generated all tags in the pattern tagName becomes tagName-ui5
Example: "ui5-button" -> ui5-button-ui5, ui5-table-row -> ui5-table-row-ui5.

Think of another way to discover if the element is a row. Perhaps you can attach a getter to the Row component that returns always true (get isRow() -> true).

Reply:
How about using `includes()`:
```js
const isRowFocused = this._getActiveElementTagName().includes("ui5-table-row");
```

Reply:
It will work for now, but is there is room for error and causes complications

var1: if we extend the the TableRow -> MyTableRow and name it ui5-sfsf-row, the code would fail
var2: if we create another class in future with tag ui5-table-row-item or something, the code would fail

Reply:
I did it with the attribute name - `this.currentItem.hasAttribute("ui5-table-row") `
This should only fail if the actual class gets renamed, or not? 

If I'm wrong there is also the option to use the currentElement getter - `this.currentElement().localname ==="tr"` `

Then again, we always can add getters in ui5-table-row and ui5-table-cell as you said, but I rather not if there is another ways to determine what we need.

Reply:
We hit exactly this case with `this.currentItem.hasAttribute("ui5-table-row")`. We extended the Table, TableColumn, TableRow and gave new tag names - `ui5-mdk-table`,` ui5-mkd-row`, etc. and this code failed, so this is again not a good solution.

I have never tried `this.currentElement().localname`, if it works and if it's supported, it seems better than the previous options.